core/src/lib.rs

```diff
@@ -14,23 +23,36 @@ pub trait DIDResolver {
     ) -> Result<(), Box<dyn std::error::Error>>;
 }
 
-/// Given the credential type and the credential subject information, create a unissued JSON-LD credential.
-/// In order to become a Verifiable Credential, a data integrity proof must be created for the credential and appended to the JSON-LD document.
-pub fn create_credential(
-    _cred_type: &str,
-    _cred_subject: serde_json::Value,
-) -> Result<serde_json::Value, Box<dyn std::error::Error>> {
-    unimplemented!();
-}
+pub trait DocumentBuilder {
+    /// Given the credential type and the credential subject information, create a unissued JSON-LD credential.
+    /// In order to become a Verifiable Credential, a data integrity proof must be created for the credential and appended to the JSON-LD document.
+    /// this is the default implementation of the `create` method. The `create` method can be overridden to create a custom credential.
+    pub fn create_credential(
+        &self,
+        cred_type: String, 
+        cred_subject: HashMap<String, Value>,
```

Comment:
Why not require the library consumer to pass something of type `CredentialSubject` here?

Reply:
It's just for consistency. The same way we are handling property_set that will be unpacked in VerifiableCredential struct, the same way we are accepting key-value to be unpacked in CredentialSubject struct. In both cases we are accepting key-value sets to unpack them later in corresponding structures